modules/ssl/ssl_engine_kernel.c

```diff
@@ -126,21 +126,62 @@ static int ap_array_same_str_set(apr_array_header_t *s1, apr_array_header_t *s2)
     return 1;
 }
 
-static int ssl_pk_server_compatible(modssl_pk_server_t *pks1, 
-                                    modssl_pk_server_t *pks2) 
+#ifdef HAVE_SSL_CONF_CMD
+/* Writes a hash of the ssl_ctx_param array to digest. */
+static void hash_ssl_params(apr_array_header_t *a,
+                            unsigned char digest[APR_MD5_DIGESTSIZE])
 {
-    if (!pks1 || !pks2) {
-        return 0;
+    apr_md5_ctx_t ctx;
+    int n;
+
+    apr_md5_init(&ctx);
+    for (n = 0; n < a->nelts; n++) {
+        ssl_ctx_param_t *p = &APR_ARRAY_IDX(a, n, ssl_ctx_param_t);
+
+        apr_md5_update(&ctx, p->name, strlen(p->name));
+        apr_md5_update(&ctx, "##", 2);
+        apr_md5_update(&ctx, p->value, strlen(p->value));
```

Comment:
Does a different ordering of the same key value pairs in the array mean a different configuration?
Won't the hashes be different if the arrays are the same but have a different ordering of the key value pairs?
Hence wouldn't we need to sort the arrays first prior to hashing? e.g.

1. Transform the original array into an array that contains `name + '##' + value` as strings
2. Sort the new array
3. Hash the new array as above